db/system_keyspace.cc

```diff
@@ -153,6 +153,7 @@ schema_ptr system_keyspace::hints() {
        builder.set_gc_grace_seconds(0);
        builder.set_compaction_strategy_options({{ "enabled", "false" }});
        builder.with_version(generate_schema_version(builder.uuid()));
+       builder.set_wait_for_sync_to_commitlog(true);
```

Comment:
Not sure about system.hints. This table can be hit heavily and it impacts user requests. @vladzcloudius @avikivity @gleb-cloudius Do we rely on hints being persisted?

@vladzcloudius btw, should we add hints to the table set returned by `maybe_write_in_user_memory()`? This affects which dirty manager is used. By default, tables from the system keyspace use the system dirty manager, which has only 10MB of space. This will cause frequent memtable flushes. maybe_write_in_user_memory() indicates that the general dirty manager should be used. We already put batchlog there, for example. hints seem like a candidate too, since the load comes from user requests.

Reply:
Do we use this table at all? We store hints in a special instance of a commitlog.

Reply:
We don't use it any more.

Reply:
> Not sure about system.hints. This table can be hit heavily and it impacts user requests. 
> @vladzcloudius @avikivity @gleb-cloudius Do we rely on hints being persisted?

@tgrabiec Our HH implementation doesn't use this table.

Reply:
So now there's a helper for creating `hints()` and `legacy::hints()`, which both are deprecates. Are there any reasons to keep them?